# Interpretation (Nestle1904LFT)

**This is work in progress**

## Table of content <a class="anchor" id="TOC"></a>
* <a href="#bullet1">1 - Introduction</a>
* <a href="#bullet2">2 - Load Text-Fabric app and data</a>
* <a href="#bullet3">3 - Examine some cases</a>
     * <a href="#bullet3x1">3.1 - Louw Nida and John 1:1 (Ἐν ἀρχῇ)</a>
     * <a href="#bullet3x2">3.2 - Formal and functional tags (ἔρχεται)</a>
     * <a href="#bullet3x3">3.3 - Handling of multiple appositions</a>

# 1 - Introduction <a class="anchor" id="bullet1"></a>
##### [Back to TOC](#TOC)

This Jupyter Notebook shows a number of cases where the interpretation of the Greek text involves some kind of choice that had to be made. However, it does not necessarily claim the data is 'wrong'.

# 2 - Load Text-Fabric app and data <a class="anchor" id="bullet2"></a>
##### [Back to TOC](#TOC)

For this notebook two data sets of Text-Fabric will be loaded. As consequence the option `hoist=globals()` can not be used and the Advanced API calls require to be called in a differnt manner. 

In [1]:
%load_ext autoreload
%autoreload 2

In [3]:
# Loading the Text-Fabric code
# Note: it is assumed Text-Fabric is installed in your environment
from tf.fabric import Fabric
from tf.app import use

In [6]:
# load the N1904GBI app and data
N1904GBI = use ("tonyjurg/Nestle1904GBI", version="0.4")

**Locating corpus resources ...**

The requested app is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904GBI/app not found


The requested data is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4 not found


   |     0.19s T otype                from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     1.95s T oslots               from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     0.59s T word                 from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     0.49s T after                from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     0.59s T book                 from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     0.51s T chapter              from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |     0.51s T verse                from ~/text-fabric-data/github/tonyjurg/Nestle1904GBI/tf/0.4
   |      |     0.05s C __levels__           from otype, oslots, otext
   |      |     1.67s C __order__            from otype, oslots, __levels__
   |      |     0.07s C __rank__             from otype, __order__
   |      |     2.21s C __levUp__            from otype, oslots, __rank__
   |      |     1.4

In [4]:
# load the N1904LFT app and data
N1904LFT = use ("tonyjurg/Nestle1904LFT", version="0.6")

**Locating corpus resources ...**

The requested app is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904LFT/app not found


The requested data is not available offline
	~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6 not found


   |     0.21s T otype                from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     2.27s T oslots               from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.63s T unicode              from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.59s T wordtranslit         from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.60s T normalized           from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.49s T chapter              from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.49s T verse                from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.59s T word                 from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.50s T after                from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0.62s T wordunacc            from ~/text-fabric-data/github/tonyjurg/Nestle1904LFT/tf/0.6
   |     0

In [7]:
# The following will push the Text-Fabric stylesheet to this notebook (to facilitate proper display with notebook viewer)
# The stylesheets of N1904GBI and N1904LFT will be identical, so one call will suffice.
N1904GBI.dh(N1904GBI.getCss())

In [8]:
# Set default view in a way to limit noise as much as possible.
# These settings need to be done for both datasets
N1904GBI.displaySetup(condensed=True, multiFeatures=False,queryFeatures=False)
N1904LFT.displaySetup(condensed=True, multiFeatures=False,queryFeatures=False)

# 3 - Performing the queries <a class="anchor" id="bullet3"></a>
##### [Back to TOC](#TOC)

## 3.1 - Louw Nida and John 1:1 (Ἐν ἀρχῇ)<a class="anchor" id="bullet3x1"></a>
##### [Back to TOC](#TOC)

Examin the Greek ἀρχῇ in John 1:1:

In [10]:
WordQuery = '''
word book=John chapter=1 verse=1 word=ἀρχῇ
'''
WordResult = N1904LFT.search(WordQuery) 
api=N1904LFT.api
# returns list of ordered tuples, even when only containing one node, hence indexed with [0][0] on next line
print("word:",api.T.text(WordResult[0][0])," Louw-Nida:",api.F.ln.v(WordResult[0][0]))

  0.08s 1 result
word: ἀρχῇ   Louw-Nida: 67.65


However, the ἀρχή of John 1:1, according to Louw-Nida Lexicon can be either:
>```
a:beginning (aspect)=68.1  [^1]
b:beginning (time)=67.65   [^2]

Only the second meaning is provided by the feature 'ln'. Although the most appropriate, the meaning of the second one (aspect) might also be implied by the author. Queries depending on 'ln' (or, by consequence, 'lex_dom') may yield incorrect or incomplete results.

[^1] Louw, Johannes P., and Eugene Albert Nida. *Greek-English Lexicon of the New Testament: Based on Semantic Domains, Vol. I* (New York: United Bible Societies, 1996), 654.

[^2] ibid., 636.

## 3.2 - Formal and functional tags (ἔρχεται)<a class="anchor" id="bullet3x2"></a>
##### [Back to TOC](#TOC)

Functional tags are only available on GBI, so the data will be taken from that Text-Fabric dataset. See [this Jupyter notebook](https://github.com/tonyjurg/Nestle1904GBI/blob/main/docs/usecases/formal_versus_functional_tag.ipynb) for an indepth discussion about formal and functional tags.

Consider the two following verses, both containing the same verb ἔρχεται:
> Καὶ ἐξῆλθεν ἐκεῖθεν, καὶ **ἔρχεται** εἰς τὴν πατρίδα αὐτοῦ, .. (He went away from there and **came** to his hometown, Mark 6:1a, ESV) 

> Ἰδοὺ **ἔρχεται** μετὰ τῶν νεφελῶν, καὶ ὄψεται αὐτὸν πᾶς ὀφθαλμὸς ... (Behold, **he is coming** with the clouds, and every eye will see him, Rev. 1:7a ESV)

The following query pulls the relevant data from the GBI Text-Fabric dataset

In [12]:
ErgetaiQuery = '''
word word=ἔρχεται
/with/
book=Mark chapter=6 verse=1 
/or/
book=Revelation chapter=1 verse=7 
/-/
'''

ErgetaiResult = N1904GBI.search(ErgetaiQuery) 
# returns list of ordered tuples

# We also need to add the app reference (N1904GBI) before we can access the F API functions
api=N1904GBI.api
for tuple in ErgetaiResult:
    node=tuple[0]
    print(api.T.sectionFromNode(node), " word:",api.T.text(node)," Formal tag:",api.F.formaltag.v(node),
      " Functional tag ",api.F.functionaltag.v(node)," gloss:",api.F.gloss.v(node))

  0.08s 2 results
('Mark', 6, 1)  word: ἔρχεται   Formal tag: V-PNI-3S  Functional tag  V-PNI-3S  gloss: came
('Revelation', 1, 7)  word: ἔρχεται   Formal tag: V-PNI-3S  Functional tag  V-PNI-3S  gloss: He is coming


Both instances are the same praesens indicativus of ἔρχομαι, with identical formal and functional tag, but require to be translated rather different. 

## 3.3 - Handling of multiple appositions<a class="anchor" id="bullet3x3"></a>
##### [Back to TOC](#TOC)

Consider the following text from Matthew 1:1:

> Βίβλος γενέσεως Ἰησοῦ Χριστοῦ υἱοῦ Δαυεὶδ υἱοῦ Ἀβραάμ. 

In this verse there are two appositions to 'Ἰησοῦ Χριστοῦ': 'υἱοῦ Δαυεὶδ' and 'υἱοῦ Ἀβραάμ'. The LFT Text-Fabric data presents 'υἱοῦ Δαυεὶδ' as apposition to 'Ἰησοῦ Χριστοῦ' and 'υἱοῦ Ἀβραάμ' as apposition to 'Ἰησοῦ Χριστοῦ υἱοῦ Δαυεὶδ'. Another choice of apposition can also be argued: both 'υἱοῦ Δαυεὶδ' and 'υἱοῦ Ἀβραάμ' being appositions to 'Ἰησοῦ Χριστοῦ'. This actualy the case in the XML data for the GBI nodes.

This Example is discussed in detail in a separate [Jupiter NoteBook](https://github.com/tonyjurg/Nestle1904LFT/blob/main/docs/usecases/appostions.ipynb)

## 3.4 Adverbial to what?<a class="anchor" id="bullet3x4"></a>
##### [Back to TOC](#TOC)

To what is the κατὰ in 1 Cor. 16:2 the adverbial? The Text-Farbic data suggest to 'μίαν σαββάτου'.

In [15]:
# create the query template
VerseQuery = '''
verse verse=2 chapter=16 book=I_Corinthians
'''
# execute the query template
VerseResult = N1904LFT.search(VerseQuery) 

  0.00s 1 result


In [16]:
N1904LFT.show(VerseResult, condensed=True, multiFeatures=False)

Query for following syntactical construction:

In [18]:
# create the query template
SyntacticQuery1 = '''
verse
  wg wgrule=PrepNp
     word lemma=κατά sp=prep
     word lemma=εἷς case=accusative
'''
SyntacticResult1 = N1904LFT.search(SyntacticQuery1) 
N1904LFT.table(SyntacticResult1, condensed=True, multiFeatures=False)

  0.21s 6 results


Query for next syntactical construction:   {note below: in the recent dataset the function of feature wgtype has changed - this query needs to be reconsidered}

In [20]:
# create the query template
SyntacticQuery2 = '''
verse
  wg wgrule=PrepNp wgrole=adv
     word sp=prep lemma=κατά
     word sp=adj case=accusative 
     wg wgtype=modifier-scope
        word sp=noun 
'''
SyntacticResult2 = N1904LFT.search(SyntacticQuery2) 
N1904LFT.table(SyntacticResult2, condensed=True, multiFeatures=False)

  0.35s 0 results


In [21]:
# create the query template
SyntacticQuery3 = '''
verse
  wg 
     word sp=prep lex_dom=067002
     word lemma=εἷς case=accusative
'''
SyntacticResult3 = N1904LFT.search(SyntacticQuery3) 
N1904LFT.table(SyntacticResult3, condensed=True, multiFeatures=False)

  0.18s 10 results


In [22]:
# create the query template
SyntacticQuery4 = '''
verse
  wg 
     a:word sp=adj lemma=εἷς
     b:word sp=conj
     c:word sp=adj
a <: b
b <: c
'''
SyntacticResult4 = N1904LFT.search(SyntacticQuery4) 
N1904LFT.table(SyntacticResult4, condensed=True, multiFeatures=False)

  0.25s 8 results


In [24]:
WordQuery = '''
word book=Titus chapter=3 verse=10 word
'''
VerseResult = N1904LFT.search(WordQuery) 
N1904LFT.show(VerseResult, condensed=True, multiFeatures=False)

  0.07s 8 results


In [25]:
api.S.relationsLegend()

                      = left equal to right (as node)
                      # left unequal to right (as node)
                      < left before right (in canonical node ordering)
                      > left after right (in canonical node ordering)
                     == left occupies same slots as right
                     && left has overlapping slots with right
                     ## left and right do not have the same slot set
                     || left and right do not have common slots
                     [[ left embeds right
                     ]] left embedded in right
                     << left completely before right
                     >> left completely after right
                     =: left and right start at the same slot
                     := left and right end at the same slot
                     :: left and right start and end at the same slot
                     <: left immediately before right
                     :> left immediately after right
   